packages/worker/src/api/controllers/admin/configs.js

```diff
@@ -16,7 +16,7 @@ exports.save = async function (ctx) {
 
   // Config does not exist yet
   if (!ctx.request.body._id) {
-    ctx.request.body._id = generateConfigID({
+    config._id = generateConfigID({
```

Comment:
When I was doing the OAuth stuff, I updated the structure of this so that it matches the joi validation. The config docs were being sent in a different format than they were being saved, and the actual configuration fields were alongside the rest of the fields in the document. The shape of the doc is now like this:

```
{
  _id: 123,
 type: "google",
 config: {
   something: 123
 }
}
```

Reply:
Yes, I was thinking about this and how to manage it, for Joi to do conditionals it had to have the type somewhere in the structure and the config as part of a structure we could have alternatives for (e.g. an array or object).

I was thinking that the best way to manage this going forward rather than requiring every read of the config do something like `const { config } = await determineScopedConfig(...)` we could just put everything into the actual config structure, I'll move the type into it too (I didn't originally but I think it should be in there too) so that the structure that is saved is like:
```
{
  _id: 123,
 type: "google",
 something: 123
}
```

basically doing `ctx.request.body = { ...ctx.request.body.config, ...ctx.request.body}; delete ctx.request.body.config`.